components/script/dom/xmlhttprequest.rs

```diff
@@ -227,10 +227,11 @@ impl XMLHttpRequest {
                 // todo
             }
 
-            fn process_response(&mut self, metadata: Result<Metadata, NetworkError>) {
+            fn process_response(&mut self,
+                                metadata: Result<Metadata, NetworkError>,
+                                _: Option<Result<Metadata, NetworkError>>) {
                 let xhr = self.xhr.root();
-                let rv = xhr.process_headers_available(self.gen_id,
-                                                       metadata);
+                let rv = xhr.process_headers_available(self.gen_id, metadata);
```

Comment:
In my proposed design, this can be:
```rust
let metadata = metadata.map(|m| match m {
    FetchResponse::Unfiltered(m) => m,
    FetchResponse::Filtered(f) => f.safe_response(),
});
```